monitor/main.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+const (
+	envAPIBaseURL     = "API_BASE_URL"
+	envAPIUsername    = "API_USERNAME"
+	envAPIAppPassword = "API_APP_PASSWORD"
+	envAPIWorkspace   = "API_WORKSPACE"
+
+	envAwsRegion          = "AWS_REGION"
+	envAwsAccessKeyID     = "AWS_ACCESS_KEY_ID"
+	envAwsSecretAccessKey = "AWS_SECRET_ACCESS_KEY"
+	envSesCharset         = "SES_CHARSET"
+	envSesReturnToAddress = "SES_RETURN_TO_ADDRESS"
+	envSesRecipientEmails = "SES_RECIPIENT_EMAILS"
+)
+
+type lambdaConfig struct {
+	BitbucketAPI bitbucketAPI `json:"BitbucketAPI"`
+	MailConfig   mail         `json:"MailConfig"`
+	Debug        bool         `json:"Debug"`
+}
+
+func (c *lambdaConfig) init() error {
+	if err := getRequiredString(envAPIBaseURL, &c.BitbucketAPI.BaseURL); err != nil {
+		return err
+	}
+
+	if err := getRequiredString(envAPIWorkspace, &c.BitbucketAPI.Workspace); err != nil {
+		return err
+	}
+	workspaceMembersURLPath = strings.ReplaceAll(workspaceMembersURLPath, "{workspace}", c.BitbucketAPI.Workspace)
+
+	if err := getRequiredString(envAPIUsername, &c.BitbucketAPI.Username); err != nil {
+		return err
+	}
+
+	if err := getRequiredString(envAPIAppPassword, &c.BitbucketAPI.AppPassword); err != nil {
+		return err
+	}
+
+	if err := getRequiredString(envSesCharset, &c.MailConfig.CharSet); err != nil {
+		return err
+	}
+
+	if err := getRequiredString(envSesReturnToAddress, &c.MailConfig.ReturnToAddr); err != nil {
+		return err
+	}
+
+	var emails string
+	if err := getRequiredString(envSesRecipientEmails, &emails); err != nil {
+		return err
+	}
+	c.MailConfig.RecipientEmails = strings.Split(emails, " ")
+
+	if err := getRequiredString(envAwsRegion, &c.MailConfig.AWSRegion); err != nil {
+		return err
+	}
+
+	if err := getRequiredString(envAwsAccessKeyID, &c.MailConfig.AWSAccessKeyID); err != nil {
+		return err
+	}
+
+	if err := getRequiredString(envAwsSecretAccessKey, &c.MailConfig.AWSSecretAccessKey); err != nil {
+		return err
+	}
+
+	var err error
+	debug := os.Getenv("DEBUG")
+	c.Debug, err = strconv.ParseBool(debug)
+	if err != nil {
+		c.Debug = false
+	}
+
+	return nil
+}
+
+func getRequiredString(envKey string, configEntry *string) error {
+	if *configEntry != "" {
+		return nil
+	}
+
+	value := os.Getenv(envKey)
+	if value == "" {
+		return fmt.Errorf("required value missing for environment variable %s", envKey)
```

Comment:
Maybe this could be a `panic` (or `os.Exit` with a write to stderr) to make it a bit simpler, not needing to return `err`. `panic` is generally avoided but this is only called from `init` so it's going to exit the program anyway.